app/main.py

```diff
@@ -4,90 +4,152 @@
 import base64
 import json
 import panel as pn
+from openvisuspy import SetupLogger, Slice, ProbeTool, cbool
+import time
 
-from openvisuspy import SetupLogger, Slice, ProbeTool,cbool
+# Global synchronization lock and throttle variables
+sync_in_progress = False
+last_sync_time = 0
+throttle_delay = 0.1  # Throttle delay in seconds
 
 # ////////////////////////////////////////////////////////////////////////////
-def SyncSlices(slice1, slice2, box1=None, box2=None):
-
-	if box1 is None: 
-		box1=slice1.db.getPhysicBox()
-
-	if box2 is None:
-		box2=slice2.db.getPhysicBox()
 
-	# map box1 to box2
-	(a, b), (c, d)=box1
-	(A, B), (C, D)=box2
+def throttle(func, *args, **kwargs):
+    """Throttle function calls to limit frequency."""
+    global last_sync_time
+    current_time = time.time()
+    if current_time - last_sync_time >= throttle_delay:
+        last_sync_time = current_time
+        func(*args, **kwargs)
 
-	x, y, w, h = slice1.canvas.getViewport()
-	x1, x2 = [A + ((value - a) / (b - a)) * (B - A) for value in [x, x + w]]
-	y1, y2 = [C + ((value - c) / (d - c)) * (D - C) for value in [y, y + h]]
-	slice2.canvas.setViewport([x1, y1, x2 - x1, y2 - y1])
-	slice2.refresh("SyncSlices")
+def SyncSlices(slice1, slice2, box1=None, box2=None):

```

Comment:
can you merge functions to avoid replication of code (syncSlices and reverse)?
also no to use global... you can use variables inside the class (i.e. if there are two couples of synced-images it will not work)

Reply:
syncSlices and reverse are merged. removed global variables and included them in the class.